### PR TITLE
Refresh plugin for June 2023

### PR DESCRIPTION
**Pull request by basil**
Closes #70.

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -2,6 +2,6 @@
   <extension>
     <groupId>io.jenkins.tools.incrementals</groupId>
     <artifactId>git-changelist-maven-extension</artifactId>
-    <version>1.4</version>
+    <version>1.6</version>
```
Comment (basil) [inline .mvn/extensions.xml:5]: Latest version at the time of this writing.

File: Jenkinsfile [hunk 2]
```diff
@@ -1 +1,4 @@
-buildPlugin(useContainerAgent: true, failFast: false, platforms: ['linux'])
+buildPlugin(useContainerAgent: true, failFast: false, configurations: [
+  [platform: 'linux', jdk: 17],
+  [platform: 'windows', jdk: 11],
```
Comment (basil) [inline Jenkinsfile:3]: The default Java version for Linux is Java 8, which no longer works on recent Jenkins cores. This change follows the recommended configuration in the archetype by adding a Linux Java 17 build and a Windows Java 11 build.

File: pom.xml [hunk 3]
```diff
@@ -6,7 +6,7 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>4.49</version>
+        <version>4.67</version>
```
Comment (basil) [inline pom.xml:9]: Latest version at the time of this writing.

File: pom.xml [hunk 4]
```diff
@@ -36,7 +36,7 @@
     <properties>
         <revision>2.6.6</revision>
         <changelist>-SNAPSHOT</changelist>
-        <jenkins.version>2.332.4</jenkins.version>
+        <jenkins.version>2.375.3</jenkins.version>
```
Comment (basil) [inline pom.xml:39]: Oldest LTS line supported by BOM at the time of this writing.

File: pom.xml [hunk 5]
```diff
@@ -63,8 +63,8 @@
         <dependencies>
             <dependency>
                 <groupId>io.jenkins.tools.bom</groupId>
-                <artifactId>bom-2.332.x</artifactId>
-                <version>1763.v092b_8980a_f5e</version>
+                <artifactId>bom-2.375.x</artifactId>
+                <version>2102.v854b_fec19c92</version>
```
Comment (basil) [inline pom.xml:67]: Latest version at the time of this writing.

File: pom.xml [hunk 6]
```diff
@@ -243,6 +243,30 @@
...
+                        <configuration>
+                            <!-- TODO fails on CI on Windows for inscrutable reasons -->
+                            <excludes>
+                                <exclude>org.biouno.unochoice.UiAcceptanceTest</exclude>
+                            </excludes>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
```
Comment (basil) [inline pom.xml:269]: Could not get this test to work on Windows, but this PR is a strict improvement over the status quo as it at least gets the other tests running on WIndows when they were not running before.

**Review by kinow — APPROVED** [review]
Code changes look good. Build is passing. Thank you very much for the PR and for the comments that made it a lot easier to review (especially since I'm completely out of the loop of the Java/JDK requirements for Jenkins). LGTM

**Comment by basil** [comment]
You're welcome! BTW, I wanted to take this opportunity to remind you that as a long-time contributor, you are eligible to vote in the upcoming Board and Officer Elections: https://community.jenkins.io/g/election-voter-2023

**Comment by kinow** [comment]
> You're welcome! BTW, I wanted to take this opportunity to remind you that as a long-time contributor, you are eligible to vote in the upcoming Board and Officer Elections: https://community.jenkins.io/g/election-voter-2023

Thank you @basil! I will abstain from voting for now as I am completely out of the loop, but will keep that in mind if my situation changes and I join a company that uses Jenkins and I where I have more time to work on plugins and follow the community & core development closely again :bow:

**Pull request closed without merging**